docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+SidechainXChainTransfer
+SidechainXChainClaim
+SidechainXChainAccountCreate
+SidechainAddWitness
+SidechainModify
+
+## List of new ledger objects
+
+Sidechain
+CrosschainSeqNum
+XChainAccountCreateState
+XChainTransferRewardState
+
+## Cross-chain transfers overview
+
+A cross-chain transfer moves assets from the mainchain to the sidechain or
+returns those assets from the sidechain back to the mainchain. Cross-chain
+transfers need a couple primitives:
+
+1) Put assets into trust on the mainchain.
+
+2) Issue wrapped assets on the sidechain.
+
+3) Return or destroy the wrapped assets on the sidechain.
+
+4) On the sidechain, prove that assets were put into trust on the mainchain.
+
+5) On the mainchain, prove that assets were returned or destroyed.
+
+6) A way to prevent assets from being wrapped multiple times (prevent
+transaction replay). The proofs that certain events happened on the different
+chains are public and can be submitted multiple times. This can only wrap (or
+unlock) assets once.
+
+In this implementation, a regular XRP ledger account is used to put assets into
+trust on the mainchain, and a regular XRP ledger account is used to issue assets
+on the sidechain. These accounts will have their master keys disabled and funds
+will move from these accounts through a new set of transactions specifically
+meant for cross-chain transfers. A special set of "witness servers" are used to
+prove that assets were put into trust on the mainchain or returned on the
+sidechain. A new ledger object called a "cross-chain sequence number" is used to
+prevent transaction replay.
+
+A source chain is the chain where the cross-chain transfer start - by either
+putting assets into trust on the mainchain or returning wrapped assets on the
+sidechain. The steps for moving funds from a source chain to a destination chain
+are:
+
+1) On the destination chain, an account submits a transaction that adds a ledger
+   object that will be used to identify the initiating transaction and prevent
+   the initiating transaction from being claimed on the destination chain more
+   than once. The door account will keep a new ledger object - a sidechain. This
+   sidechain ledger object will keep a counter that is used for "cross chain
+   sequence numbers". A "cross chain sequence number" will be checked out from
+   this counter and the counter will be incremented. Once checked out, the
+   sequence number would be owned by the account that submitted the transaction.
+   See the section below for what fields are present in the new sidechain ledger
+   object and cross chain sequence number ledger object. The actual number must
+   be retrieved from the transaction metadata on a validated ledger.
+   
+2) On the source chain, an initiating transaction is sent from a source account.
+   This transaction will include the sidechain, "cross chain sequence number"
+   from step (1), a signature reward amount, in XRP, and an optional destination
```

Comment:
Why does the reward amount need to be specified, if it needs to be the same as in the Sidechain object?

Reply:
Because we're taking funds out of sending's account's funds, I don't want this to be invisible. I want them to positively say "yes, I know I'm paying this amount".